pkg/controllers/externaloidc/generation/common.go

```
@@ -0,0 +1,235 @@
+	wellKnown := strings.TrimSuffix(hostURL, "/")
+	req, err := http.NewRequest(http.MethodGet, wellKnown, nil)
+	if err != nil {
+		return fmt.Errorf("could not create well-known HTTP request: %v", err)
+	}
+
+	var resp *http.Response
+	var connErr error
+	retryCtx, cancel := context.WithTimeout(req.Context(), 10*time.Second)
+	defer cancel()
+	retry.RetryOnConnectionErrors(retryCtx, func(ctx context.Context) (done bool, err error) {
+		resp, connErr = client.Do(req)
```

golangci-lint (2.12.2)

[error] response body must be closed

(bodyclose)